src/main.c

```diff
@@ -534,11 +534,29 @@ void dict(struct state *s)
 			}
 		} else if (strcmp(subfunc, "auth") == 0) {
 			if (peek_arg(&(s->cmd_list)) == NULL) {
-				printf("No username specified.\n");
-			} else {
+				char filename[] = "user.inf";
+				FILE*f;
+				int len;
+				rsp = exword_get_file(s->device, basename(filename), &buffer, &len);
+				if (rsp == 0x20)
+					rsp = write_file(filename, buffer, len);
```

Comment:
There should be no need to write the buffer to disk only to read it back in. It would be better to just use sscanf on the buffer retrieved from exword_get_file if it succeeds.

Reply:
Do you mean that just use `exword_get_file` and `sscanf` rather than file method to get auth user name?

Reply:
Yes, there should not be any need to write the buffer returned from exword_get_file to disk only to read it back in via fscanf. Instead you should just be able to use sscanf on the buffer directly.